packages/graphql-transformer-core/src/util/amplifyUtils.ts

```diff
@@ -113,29 +113,52 @@ function mergeUserConfigWithTransformOutput(
             return resourceTypesToDependOn[resource.Type];
         }
     );
+    // Looping through the parameters defined by the transform (aka. rootStack)
     const parametersKeys = Object.keys(rootStack.Parameters);
     const customStackParams = parametersKeys.reduce((acc: any, k: string) => ({
         ...acc,
         [k]: Fn.Ref(k)
     }), {})
+    // customStackParams is a map that will be passed as the "parameters" value
+    // to any nested stacks.
     customStackParams[ResourceConstants.PARAMETERS.AppSyncApiId] = Fn.GetAtt(
         ResourceConstants.RESOURCES.GraphQLAPILogicalID,
         'ApiId'
     );
 
+    // Load the root stack's parameters as we will update them with the Child Stack's parameters
+    // if they are not already present in the root stack.
+    let updatedParameters = rootStack.Parameters;
+
     for (const userStack of Object.keys(userStacks)) {
         if (transformOutput.stacks[userStack]) {
             throw new Error(`You cannot provide a stack named ${userStack} as it \
             will be overwritten by a stack generated by the GraphQL Transform.`)
         }
         const userDefinedStack = userConfig.stacks[userStack];
+
+        /**
+         * First loop through the parameters in the user defined stack and see
+         * if there are any Parameters that are present in the child but not the
+         * root stack - if so, add it to the root stack.
+         */
+        const parameterKeys = Object.keys(userDefinedStack.Parameters);
+
+        for (let i = 0; i < parameterKeys.length; i += 1) {
+            if (customStackParams[parameterKeys[i]] == null) {
+                customStackParams[parameterKeys[i]] = Fn.Ref(parameterKeys[i]);
+                // Add the entire parameter entry from the user defined stack's parameter
+                updatedParameters[parameterKeys[i]] = userDefinedStack.Parameters[parameterKeys[i]];
```

Comment:
I think you should check if the `updatedParameters` already has a parameter associated with that key and fail with a helpful error if they do. We should enforce that parameter names are unique across the project, and users should not be able to overwrite parameters that are required by the library.


```javascript
if (updatedParameters[parameterKeys[i]]) {
  throw new Error(`Cannot redefine CloudFormation parameter ${parameterKeys[i]} in stack ${stackName}.`);
} else {
  updatedParameters[parameterKeys[i]] = userDefinedStack.Parameters[parameterKeys[i]];
}
```

Reply:
done